src/main/java/com/pusher/client/channel/impl/PrivateEncryptedChannelImpl.java

```diff
@@ -0,0 +1,115 @@
+package com.pusher.client.channel.impl;
+
+import com.pusher.client.AuthorizationFailureException;
+import com.pusher.client.Authorizer;
+import com.pusher.client.channel.ChannelState;
+import com.pusher.client.channel.PrivateEncryptedChannel;
+import com.pusher.client.channel.PrivateEncryptedChannelEventListener;
+import com.pusher.client.channel.SubscriptionEventListener;
+import com.pusher.client.connection.impl.InternalConnection;
+import com.pusher.client.crypto.nacl.SecretBoxOpener;
+import com.pusher.client.util.Factory;
+import com.pusher.client.util.internal.Base64;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class PrivateEncryptedChannelImpl extends ChannelImpl implements PrivateEncryptedChannel {
+
+    private final InternalConnection connection;
+    private final Authorizer authorizer;
+    protected SecretBoxOpener secretBoxOpener;
```

Comment:
There is [that previous comment](https://github.com/pusher/pusher-websocket-java/pull/234#commitcomment-38165917) which hasn't been addressed.

Reply:
Could you maybe resolve some of the existing conversations like we agreed? It's making it harder to keep track of work is still remaining.

It looks like we don't have have a SecretBoxOpenerFactory - I'll work on adding that part now. Is everything else acceptable to be merged in?

Reply:
Have now implemented a SecretBoxOpenerFactory with some advice on a video call.

Reply:
> Could you maybe resolve some of the existing conversations like we agreed?

The agreement was that only author of the feedback can be marking their feedback as resolved (as opposed to have to expand everything when revisiting a PR). I wasn't aware it would be also helpful for you as a receiver of the feedback so I was doing it at my discretion when it was useful for me. I have now marked all the bits of feedback fully resolved as resolved. The remaining (those more minor or which were broader than the focus of this change) may be good for future reference (e.g. when considering improvements) so I left them non-collapsed.